contrib/pg_stat_statements/sql/test_pg_stat_statements.sql

```diff
@@ -0,0 +1,215 @@
+
+-- change alias name
+SELECT
+    GROUPING(item2) AS item_2_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+-- check that recent request was added to preceding entry
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check that different grouping options result in separate entries
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item1, item2, item3, item4);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item1, item2, item3, item4);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY GROUPING SETS(item1, item2, item3, item4);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY GROUPING SETS((item1, item2), (item3, item4));
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item1, item2, item3, item4;
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several parameters options in ROLLUP
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item1, item2);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item2, item3);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item3, item4);
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several parameters options in CUBE
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item1, item2);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item2, item3);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item3, item4);
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several parameters options in GROUPING SETS
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY GROUPING SETS((item1, item2), (item2, item3));
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY GROUPING SETS((item1, item2), (item3, item4));
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several options in simple GROUP BY
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item1, item2;
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item2, item3;
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item3, item4;
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item1, item2, item3;
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+SET optimizer='on';
```

Comment:
use `reset`

Reply:
Do you mean to use pg_stat_statements_reset() at the end of the test?

Reply:
i mean use `RESET optimizer;` instead of `SET optimizer='on';`